src/StatusQ/Components/private/statusMessage/StatusPinMsgComponent.qml

```diff
@@ -0,0 +1,45 @@
+import QtQuick 2.13
+import QtQuick.Layouts 1.14
+import QtGraphicalEffects 1.13
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+Loader {
+    property string pinnedBy: ""
+
+    active: visible
+
+    sourceComponent: Rectangle {
+        height: 24
+        width: layout.width + 16
+        color: Theme.palette.pinColor2
+        radius: 12
+        RowLayout {
+            id: layout
+            anchors.centerIn: parent
+            StatusIcon {
+                Layout.alignment: Qt.AlignVCenter
+                Layout.preferredWidth: 16
+                Layout.preferredHeight: 16
+                color: Theme.palette.pinColor1
+                icon: "tiny/pin"
+            }
+            StatusBaseText {
+                Layout.alignment: Qt.AlignVCenter
+                Layout.leftMargin: -4
+                color: Theme.palette.directColor1
+                font.pixelSize: 13
+                text: "Pinned by"
```

Comment:
damn yes! I think there are more like. Will expose them to the outside. However can we not support internationalisation for the statusQ components in the future?

Reply:
> However can we not support internationalisation for the statusQ components in the future?

So.. exposing the labels is actually what makes i18n possible in whatever way consumers like to do it.
The main I went down this path in StatusQ is because, if we put qsTr() calls inside components, Status Desktop's i18n script will rewrite them as qsTrId(), which essentially changes the source of this third party library.

Not sure if there are better practices/solutions available in QML land. Maybe we're doing it wrong in Status Desktop, I dunno.

But I'm happy to explore better ways. In fact @noeliaSD had some idea in this thread here: https://github.com/status-im/StatusQ/pull/535#discussion_r787704959

If you two want to explore this further, please go ahead and create an issue where the discussion happens!
That'd be awesome :D

Reply:
I think we should at least discuss and maybe create ticket for it.